compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java

```diff
@@ -6377,7 +6377,7 @@ private void rewriteFieldBasedAccess(BLangFieldBasedAccess fieldAccessExpr) {
                         (BVarSymbol) fieldAccessExpr.symbol, false, fieldAccessExpr.isStoreOnCreation);
             }
         } else if (types.isLaxFieldAccessAllowed(refType)) {
-            if (!(varRefTypeTag == TypeTags.XML || varRefTypeTag == TypeTags.XML_ELEMENT)) {
```

Comment:
There seems to be an issue in the type set for optional attribute access.

```ballerina
xml:Comment x = xml `<!--This is a comment-->`;
string? xmlResult = x?.attr; // should result in a compile-time error
io:println(<any|error> xmlResult is error); // true, although static type doesn't allow error
```

Reply:
Should this line result in a compile time error?
```ballerina
string? xmlResult = x?.attr; // should result in a compile-time error
```
I'm assuming the reason this should be a compile time error is that the static type of the right hand side should be `string|error?`.
But as per the spec, https://ballerina.io/spec/lang/master/#section_6.12 the optional attribute access results in an error when `if v is not a singleton element, the result is an error. `  if the element x here is not a singleton. But in this case we know that the element is a singleton. So should this still be a compile time error?

Reply:
That explanation is for runtime behaviour. The static type is defined simply as

> The static type of an xml-required-attribute-access-expr is string|error. The static type of an xml-optional-attribute-access-expr is string|error|(). 

So the implementation has to conform to this.

Reply:
Ack. Thanks
This seems to exists the existing implementation as well.
```ballerina
function foo() {
    xml:Element x = xml `<x/>`;
    string? xmlResult = x?.attr;
}
```
Will update the fix